README.md

```diff
@@ -13,15 +13,19 @@ git clone git@github.com:maas/maas-ansible-playbook
 ## Setup
 This playbook has several main roles, each has a corresponding group to assign hosts to. They are as follows:
 
-- **maas_postgres_primary**: This role and group is for a host to be configured as an initial Postgres primary instance for the MAAS stack. This role is required.
+- **maas_pacemaker**: This role and group are for a set of hosts to be configured as a pacemaker cluster to manage HA Postgres. It is recommended that `maas_corosync` is a child group of this group. It is also recommended that these are the same hosts as the postgres group.
 
-- **maas_postgres_secondary**: This role and group is for hosts to be configured as Postgres secondaries in hot standby with synchronous replication. This role is optional.
+- **maas_corosync**: This role and group are for a set of hosts to be configured to run a corosync cluster, used for managing quorum of a HA Postgres cluster
 
-- **maas_region_controller**: This role and group is for hosts to be configured as MAAS Region Controllers. This role is required at least once.
+- **maas_postgres**: This role and group are for a host to be configured as a Postgres instance for the MAAS stack. If more than one host is assigned this role, Postgres will be configured in HA, which then makes the `maas_corosync` and `maas_pacemaker` roles required. This role is required.
 
-- **maas_rack_controller**: This role and group is for hosts to be configured as MAAS Rack Controllers. This role is required at least once.
+- **maas_region_controller**: This role and group are for hosts to be configured as MAAS Region Controllers. This role is required at least once.
 
-- **maas_proxy**: This role and group is for hosts to be configured as an HAProxy instance in front of region controllers for HA stacks. This role is optional.
+- **maas_rack_controller**: This role and group are for hosts to be configured as MAAS Rack Controllers. This role is required at least once.
+
+- **maas_postgres_proxy**: This role and group are for hosts to be configured as a HAProxy instance in front of the HA Postgres cluster, in order to ensure queries are directed to the primary instance. We recommend this group is assigned to the same hosts as `maas_region_controller`, this will have HAProxy's listener bind to localhost and route queries for that specific region controller.
```

Comment:
as an HAProxy instance

Reply:
We should also add whether this role is optional or not

Reply:
Also suggesting a stylistic change: 'It is recommended that this group is assigned to the same hosts as `maas_region_controller`, ..' for consistency